trie-db/src/triedb.rs

```diff
@@ -142,6 +155,14 @@ where
 	> {
 		TrieDBIterator::new(self).map(|iter| Box::new(iter) as Box<_>)
 	}
+
+	fn key_iter<'a>(&'a self)-> Result<
```

Comment:
sidenote: this is not needed for the PR isn't it ? maybe the optimization could be moved to another PR, that would make review easier to me at least

Reply:
thinking again, maybe it is useful for big keys

Reply:
Main purpose of the specific key iterartor is to not access values, so we don't have values in proofs but only their hashes when unaccessed.
(that is what next_storage_key need to use in substrate)